### PR TITLE
Bump torchtune pin to a 9-24 commit; Update Flamingo Definition 

### PR DESCRIPTION
**Pull request by Gasoonjia**
Torchtune updates flamingo model definition, as well as its inference. This PR aims to update torchchat pipeline to sync with torchtune

### CODE REVIEW

**Review by Jack-Khuu — APPROVED** [review]

File: install/install_requirements.sh [hunk 1]
```diff
@@ -92,6 +87,12 @@ REQUIREMENTS_TO_INSTALL=(
   $PIP_EXECUTABLE install torchao=="0.5.0"
 )
 
+# Rely on the latest tochtune for flamingo support
+(
+  set -x
+  $PIP_EXECUTABLE install git+https://github.com/pytorch/torchtune.git@18efc81dda1c537bb7c25058ff059b4623ccff58
+)
```
Comment (Jack-Khuu) [inline install/install_requirements.sh:94]: This works on Mac now right?

File: torchchat/generate.py [hunk 2]
```diff
@@ -593,7 +604,7 @@ def generate(
                     self.is_torchtune_model
                     or self.model.config.model_type == ModelType.Flamingo
                 ):
-                    model.setup_caches(max_batch_size=1, dtype=self.dtype)
+                    model.setup_caches(batch_size=1, dtype=self.dtype, encoder_max_seq_len=6404, decoder_max_seq_len=max_seq_length-1)
```
Comment (Jack-Khuu) [inline torchchat/generate.py:607]: Magic number 6404?

**Review by Jack-Khuu — APPROVED** [review]

File: install/install_requirements.sh (hunk 1 above)
Comment (Gasoonjia) [reply install/install_requirements.sh:94]: yeah and mac tests passed

File: torchchat/generate.py (hunk 2 above)
Comment (Gasoonjia) [reply torchchat/generate.py:607]: it is the size that can hold single image input while one-gpu affordable.

Comment (Gasoonjia) [reply torchchat/generate.py:607]: I will leave a comment

**Pull request merged**